src/microsoft_demangle.c

```diff
@@ -1044,16 +1239,14 @@ DEF_STATE_ACTION(U) {
 #define PROCESS_CASE(case_string, type_str) \
 	{ \
 		check_len = strlen(case_string); \
-		if ((check_len < buff_len) && \
-			(strncmp(state->buff_for_parsing, case_string, check_len) == 0)) { \
-			copy_string(type_code_str, type_str, 0); \
+		if (strncmp(state->buff_for_parsing, case_string, check_len) == 0) { \
+			copy_string(type_code_str, type_str, strlen(type_str)); \
```

Comment:
are these changes to `copy_string` calls really necessary? I think specifying the strlen every time is very error prone and I'd prefer to have the code compute it itself. There should be some way to force the compiler to optimize the strlen when the string is fixed.

Reply:
Changed `copy_string` to a macro that does `strlen(str)` and doesn't have a length param. The compiler optimizes `strlen()` if the string is known at compile time, (which is the case in this)